### ISSUE TITLE
Consider implementing ArrayWithBasis

### ISSUE THREAD

**Issue by williamesamuelson** [williamesamuelson/Majoranas.jl#25]
Consider adding an ArrayWithBasis{T,N,B} type. Operators and states could then carry basis information, as mentioned in #24.

**Comment by cvsvensson** [comment]
Testing it out in the branch `basisarray`. One could also try with https://github.com/haberdashPI/MetaArrays.jl

**Comment by williamesamuelson** [comment]
Looks cool! Do you think this is sort of separate from #24 if we choose to go the `dict` path there?

**Comment by cvsvensson** [comment]
Yes. The convenience of BasisArray would be that we don't have to specify the basis when calling various functions, such as wedge, blockdiagonal, partial_trace...
